license-header-check/action.yml

```diff
@@ -0,0 +1,68 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: 'License Header Check'
+description: 'check license/copyright header'
+inputs:
+  included_file_patterns:
+    description: "included file pattern"
+    required: true
+    type: string
+
+runs:
+  using: "composite"
+  steps:
+    - name: Get changed files
+      shell: bash
+      run: |
+        CURRENT_YEAR=$(date +%Y)
+        LICENSE_PATTERN="Copyright \(c\) .*?${CURRENT_YEAR}, NVIDIA CORPORATION."
+        
+        # Format file patterns
+        IFS="," read -r -a FILE_PATTERN_ARRAY <<< ${{ inputs.included_file_patterns }}
+        PATTERNS=""
+        for PATTERN in "${FILE_PATTERN_ARRAY[@]}"; do
+            PATTERNS+="$(echo "$PATTERN" | xargs -I{} echo "{}") "
+        done
+
+        # Get changed files
+        FILES=$(git diff --name-only ${{ github.event.pull_request.base.sha }} ${{ github.event.pull_request.head.sha }})
```

Comment:
I wonder if we should have an optional check for all files in the repo (not just diff) showing if there are missed files from before this check has been introduced?

Reply:
this could be a one-time run before we enable the check for each repo, and then we just need to detect modified files only